flax/nn/base.py

```diff
@@ -279,6 +279,22 @@ def __new__(cls, *args, name=None, **kwargs):
 
   @abc.abstractmethod
   def apply(self, *args, **kwargs):
+    """Override to implement how a module is initialized and called.
+
+    Within `apply`, you can generally do two things:
+
+    * Call `self.param(name, shape, initializer)` to declare
```

Comment:
Perhaps the first point here should be that you can do arbitrary numerical computation using Jax